bridges/snowbridge/primitives/router/src/inbound/mod.rs

```diff
@@ -273,8 +273,10 @@ where
 			},
 			None => {
 				instructions.extend(vec![
-					// Deposit asset to beneficiary.
-					DepositAsset { assets: Definite(asset.into()), beneficiary },
+					// Deposit both asset and fees to beneficiary so the fees will not get
+					// trapped. Another benefit is when fees left more than ED on AssetHub could be
+					// used to create the beneficiary account in case it does not exist.
+					DepositAsset { assets: Wild(AllCounted(2)), beneficiary },
```

Comment:
Don't you need to fix also `DepositReserveAsset {` branch?
```
// Deposit asset to beneficiary.
DepositAsset { assets: Definite(asset.into()), beneficiary },
```

Reply:
> One question: could you point me to the place or explain where `asset_hub_fee` and `destination.fee` is coming from? Is it withdrawn or burned somewhere on BH? Is it the user's money, or is it money from some sovereign account on BH? Or where does this money come from?



My concern here is whether we should deposit unused fees directly to the beneficiary instead of depositing them into a sovereign account in the AH. If this is a valid scenario, then the process should resemble something like this:
```
DepositAsset { assets: Definite(vec![asset].into())), beneficiary},
DepositAsset { assets: Definite(vec![asset_hub_fee_asset].into())), beneficiary: LocationOfSomeSovereignAccountOnAssetHub},
```

Reply:
> where asset_hub_fee and destination.fee is coming from? 

`asset_hub_fee` from https://github.com/Snowfork/snowbridge/blob/3f90b1619d0bdbec8f50b63185b3dcc632321019/contracts/src/storage/AssetsStorage.sol#L15 which is a configurable storage on Ethereum, we charge it from the user and make sure that it can cover the execution cost on AH.

`destination.fee` is always zero for AH.

> whether we should deposit unused fees directly to the beneficiary

 I think so, it's from end user and we do want to send the unused back to the user. 
 
 > Don't you need to fix also DepositReserveAsset { branch?
 
 Nope, that branch is for non-system parachain which may not accept `AssetHub-DOT` as fee asset, we're gonna to make some fix there later and this PR focus on AH only.

Reply:
> I think so, it's from end user and we do want to send the unused back to the user.

ok, cool, thank you :)

> > Don't you need to fix also DepositReserveAsset { branch?
> 
> Nope, that branch is for non-system parachain which may not accept `AssetHub-DOT` as fee asset, we're gonna to make some fix there later and this PR focus on AH only.

Ok, got it. I saw that you use `Location::parent()` here for fees.assetId, so that would be probably changed and/or `Destination.fee: u128`, let's see later.

So, I think using exact assets for DepositAsset, as suggested above, is not a bad idea (or maybe some exact counter of unique assetIds :))